File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        length = len(nums)
        
        fetcher = {}
        
        for i in range(length):
            search = target - nums[i]
            if search in fetcher and fetcher[search]!=i:
                return [i,fetcher[search]]
            else:
                fetcher[nums[i]] = i
            
            

File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
         unordered_map<int,int> mymap;
        for(int i=0;i<nums.size();i++){
            int finder = target - nums[i];
            if(mymap.find(finder)!=mymap.end())
            {
                return {i,mymap[finder]};
            }
            mymap[nums[i]] = i;
        }
        return {};
    }
};